API/src/main/java/de/brandwatch/minianalytics/api/service/QueryService.java

```diff
@@ -0,0 +1,49 @@
+package de.brandwatch.minianalytics.api.service;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import de.brandwatch.minianalytics.api.postgres.model.Query;
+import de.brandwatch.minianalytics.api.postgres.repository.QueryRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+
+@Component
+public class QueryService {
+
+    private static final Logger logger = LoggerFactory.getLogger(QueryService.class);
+
+    private static final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+    @Autowired
+    private QueryRepository queryRepository;
+
+    public ResponseEntity createQuery(String query){
+        try {
+            Query saveQuery = new Query(query);
+
+            logger.info("Retrieved query: " + query);
+            return ResponseEntity.status(200).body(gson.toJson(queryRepository.save(saveQuery)));
```

Comment:
looking at it like this, I think it might be a little unconventional to return a ResponseEntity here instead of a query. Same of course for the other methods. Especially if you consider that you might wanna use this service for something else later. Let's assume you have a MentionController to fetch mentions for a given query, but first you want to validate if the query exists and use this Service for that.

It might be better to just throw a meaningful exception in the catch and handle the HTTP Status with some exception handling solution. There's multiple of those in Spring: https://www.baeldung.com/exception-handling-for-rest-with-spring